### PR TITLE
Disable prometheus rule evaluation if mimir is enabled

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR disable rule evaluation (rule wiring) if mimir is enabled on the installation towards https://github.com/giantswarm/roadmap/issues/3298

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Disable rule evaluation in Prometheus when Mimir is enabled.
+
+### Removed
+
+- Remove `prometheus` and `prometheus_replica` external labels when Mimir is enable.
```
Comment (QuentinBisson) [inline CHANGELOG.md:16]:
```suggestion
- Remove `prometheus` and `prometheus_replica` external labels when Mimir is enabled.
```

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM but I'm no go expert

**Review by hervenicol — COMMENTED** [review]
The removal of default external labels seems a bit sketchy to me, I'm not sure I caught all of the implications of it. :thinking: 

But LGTM, let's see how it goes.

**Review by hervenicol — APPROVED** [review]

**Comment by QuentinBisson** [comment]
@hervenicol we removed them in the prometheus agents because the shared metric was causing high cardinality so let's remove and align

**Pull request merged**